src/sql/workbench/parts/notebook/notebookActions.ts

```diff
@@ -489,14 +490,14 @@ export class AttachToDropdown extends SelectBox {
 	 * Bind the server value to 'Attach To' drop down
 	 * Connected server is displayed at the top of drop down
 	 **/
-	public async openConnectionDialog(useProfile: boolean = false): Promise<void> {
+	public async openConnectionDialog(useProfile: boolean = false): Promise<boolean> {
 		try {
 			await this._connectionDialogService.openDialogAndWait(this._connectionManagementService, { connectionType: 1, providers: this.model.getApplicableConnectionProviderIds(this.model.clientSession.kernel.name) }, useProfile ? this.model.connectionProfile : undefined).then(connection => {
```

Comment:
Minor: given this is async, can we remove the `.then` clause? That would let us just return false / true instead of having to do Promise.resolve inside the thenable.